src/NBitcoin/ChainedHeader.cs

```diff
@@ -4,6 +4,60 @@
 
 namespace NBitcoin
 {
+    /// <summary>
+    /// Represents the state of the availability of a block.
+    /// </summary>
+    public enum BlockDataAvailabilityState
+    {
+        /// <summary>
+        /// A <see cref="BlockHeader"/> is found and is a valid header.
+        /// </summary>
+        HeaderOnly,
+
+        /// <summary>
+        /// We are interested in downloading the <see cref="Block"/> that is being represented by the current <see cref="BlockHeader"/>. 
+        /// This happens when a header is a part of the chain that can potentially replace our consensus tip 
+        /// because its tip's total chain work is greater comparing to what we currently have.
+        /// </summary>
+        BlockRequired,
+
+        /// <summary>
+        /// The <see cref="Block"/> is downloaded, if the Block property is null the block needs to be asked from store.
```

Comment:
You don't really have a block property here. So actually this part of the comment should be on the actual property inside of chained header, not here. By itself this value means that the block data has been downloaded and is still reachable. It is possible that a block was downloaded by light wallet and it was analysed and it is no longer needed and therefore the block availability state will be changed to header only. But here you should not mention anything about the store of the block data.